File: Algorithm_455_Assign Cookies.cpp
#include <vector>
#include <algorithm>
using namespace std;
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int child=0, cookie=0, content=0;
        while (child < g.size() && cookie < s.size())
        {
            if (s[cookie] < g[child])//cannot be satisfied
            {
                cookie++;
            }
            else//can be satisfied
            {
                cookie++;
                child++;
                content++;
            }
        }
        return content;
    }
};

File: Algorithm_455_Assign Cookies.java
import java.util.Arrays;

// 对g和s排序，然后尽量把大的cookie assign给需要大cookie的child

class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);

        int i = g.length - 1, j = s.length - 1;
        int result = 0;
        while (i >= 0 && j >= 0) {
            if (g[i] <= s[j]) { // can assign cookie j to child i
                j--;
                result++;
            }
            i--;
        }
        return result;
    }
}